subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/internal/SourceFoldersCreator.java

```diff
@@ -126,6 +126,8 @@ public String apply(SourceFolder sourceFolder) {
                     folder.setName(dir.getName());
                     folder.setIncludes(getIncludesForTree(sourceSet, tree));
                     folder.setExcludes(getExcludesForTree(sourceSet, tree));
+
+                    folder.getEntryAttributes().put("gradle_source_sets", sourceSet.getName().replaceAll(",", ""));
```

Comment:
⭕️ Can the same directory be used in different sourceSets, with different includes/excludes? If yes, how does this algorithm handle that?

Reply:
The Gradle Eclipse plugin generates separate source folder classpath entries, even if the corresponding source sets use the same directory in the Java plugin configuration. I haven't noticed it since Buildship only [considers](https://github.com/eclipse/buildship/blob/b566a865fd6e53851f81881615bd3f82320bf109/org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/SourceFolderUpdater.java#L62-L65) the last source folder with the same path. The current version of Eclipse doesn't support multiple source folders with the same path (causes a build path error), so the Gradle Eclipse plugin is broken in this regard.
Should we fix this in this story?

Reply:
No, I just wanted to make sure we are not missing a valid case. We can fix this if it ever actually comes up as a problem.

Reply:
I don't think this is worth fixing in this story. I just wanted to make sure we were not missing an important use case.

Reply:
I see. Anyway, thanks for pointing this out, you made me double check all the possible use-cases.